go/multipooler/manager/pg_multischema.go

```diff
@@ -0,0 +1,368 @@
+	defer cancel()
+
+	_, err := pm.db.ExecContext(execCtx, "CREATE SCHEMA IF NOT EXISTS multigres")
+	if err != nil {
+		pm.logger.ErrorContext(ctx, "Failed to create multigres schema", "error", err)
+		return mterrors.Wrap(err, "failed to create multigres schema")
+	}
+
+	return nil
+}
+
+// ----------------------------------------------------------------------------
+// Table Creation
+// ----------------------------------------------------------------------------
+
+// createHeartbeatTable creates the heartbeat table for leader election
+func (pm *MultiPoolerManager) createHeartbeatTable(ctx context.Context) error {
+	execCtx, cancel := context.WithTimeout(ctx, 500*time.Millisecond)
+	defer cancel()
+
+	_, err := pm.db.ExecContext(execCtx, `
+		CREATE TABLE IF NOT EXISTS multigres.heartbeat (
+			shard_id BYTEA PRIMARY KEY,
+			leader_id TEXT NOT NULL,
+			ts BIGINT NOT NULL
+		)
+	`)
+	if err != nil {
+		pm.logger.ErrorContext(ctx, "Failed to create heartbeat table", "error", err)
+		return mterrors.Wrap(err, "failed to create heartbeat table")
+	}
+
+	return nil
+}
+
+// createDurabilityPolicyTable creates the durability_policy table and its indexes
+func (pm *MultiPoolerManager) createDurabilityPolicyTable(ctx context.Context) error {
+	execCtx, cancel := context.WithTimeout(ctx, 500*time.Millisecond)
+	defer cancel()
+
+	_, err := pm.db.ExecContext(execCtx, `
+		CREATE TABLE IF NOT EXISTS multigres.durability_policy (
+			id BIGSERIAL PRIMARY KEY,
+			policy_name TEXT NOT NULL,
+			policy_version BIGINT NOT NULL,
+			quorum_rule JSONB NOT NULL,
+			is_active BOOLEAN NOT NULL DEFAULT true,
+			created_at TIMESTAMPTZ NOT NULL DEFAULT now(),
+			updated_at TIMESTAMPTZ NOT NULL DEFAULT now(),
+			UNIQUE (policy_name, policy_version),
+			CONSTRAINT quorum_rule_required_count_check CHECK (
+				(quorum_rule->>'required_count')::int >= 1
+			)
+		)
+	`)
+	if err != nil {
+		pm.logger.ErrorContext(ctx, "Failed to create durability_policy table", "error", err)
+		return mterrors.Wrap(err, "failed to create durability_policy table")
+	}
+
+	// Create index on is_active for efficient active policy lookups
+	execCtx, cancel = context.WithTimeout(ctx, 500*time.Millisecond)
+	defer cancel()
+
+	_, err = pm.db.ExecContext(execCtx, `
+		CREATE INDEX IF NOT EXISTS idx_durability_policy_active
+		ON multigres.durability_policy(is_active)
+		WHERE is_active = true
+	`)
+	if err != nil {
+		pm.logger.ErrorContext(ctx, "Failed to create durability_policy index", "error", err)
+		return mterrors.Wrap(err, "failed to create durability_policy index")
+	}
+
+	return nil
+}
+
+// ----------------------------------------------------------------------------
+// Multischema Global Tables (default tablegroup only)
+// ----------------------------------------------------------------------------
+
+// createTablegroups creates the tablegroup table for tracking table groups
+func (pm *MultiPoolerManager) createTablegroups(ctx context.Context) error {
```

Comment:
nit: the function name `createTablegroups` does not match the table name `tablegroup`